VideoApp/VideoApp/ViewControllers/Chat/ChatViewController.swift

```diff
@@ -16,29 +16,29 @@
 
 import UIKit
 
-class ChatViewController: UIViewController {
-    private let chatStore: ChatStoreWriting = ChatStore.shared
+class ChatViewController: UITableViewController {
+    var viewModel: ChatViewModel!
     
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        NotificationCenter.default.addObserver(
-            self,
-            selector: #selector(handleChatStoreUpdate),
-            name: .chatStoreUpdate,
-            object: chatStore
-        )
-        
-        chatStore.isUserReadingMessages = true
+        viewModel.delegate = self
+        viewModel.isUserReadingMessages = true
         
-        print("Chat > All messages:")
-        chatStore.messages.forEach { print("    \($0)") }
+        tableView.register(
+            UINib(nibName: "ChatHeaderView", bundle: nil),
+            forHeaderFooterViewReuseIdentifier: "ChatHeaderView"
+        )
```

Comment:
This is ugly. After this PR is reviewed I will create a new PR that implements [this solution](https://medium.com/@gonzalezreal/ios-cell-registration-reusing-with-swift-protocol-extensions-and-generics-c5ac4fb5b75e) which looks really nice for both header views and cells. It will only take a few minutes but I thought it would add too much noise to this PR.

Reply:
The solution looks very clean and neat.